packages/did/src/DidDetails/LightDidDetails.utils.ts

```diff
@@ -16,165 +16,158 @@ import type {
     return
   }
 
   // Checks that for all service IDs have regular strings as their ID and not a full DID.
   // Plus, we forbid a service ID to be `authentication` or `encryption` as that would create confusion
   // when upgrading to a full DID.
-  details.serviceEndpoints?.forEach((service) => {
+  details.service?.forEach((service) => {
     // A service ID cannot have a reserved ID that is used for key IDs.
-    if (service.id === 'authentication' || service.id === 'encryption') {
+    if (service.id === '#authentication' || service.id === '#encryption') {
       throw new SDKErrors.DidError(
         `Cannot specify a service ID with the name "${service.id}" as it is a reserved keyword`
       )
     }
     const [, errors] = checkServiceEndpointSyntax(service)
-    if (errors && errors.length) {
+    if (errors && errors.length > 0) {
       throw errors[0]
     }
   })
 }
 
-const ENCRYPTION_KEY_MAP_KEY = 'e'
-const SERVICES_KEY_MAP_KEY = 's'
+const KEY_AGREEMENT_MAP_KEY = 'e'
+const SERVICES_MAP_KEY = 's'
 
 interface SerializableStructure {
-  [ENCRYPTION_KEY_MAP_KEY]?: NewDidEncryptionKey
-  [SERVICES_KEY_MAP_KEY]?: DidServiceEndpoint[]
+  [KEY_AGREEMENT_MAP_KEY]?: NewDidEncryptionKey
+  [SERVICES_MAP_KEY]?: Array<Omit<DidServiceEndpoint, 'id'> & { id: string }>
 }
 
 /**
  * Serialize the optional encryption key of an off-chain DID using the CBOR serialization algorithm
  * and encoding the result in Base58 format with a multibase prefix.
  *
  * @param details The light DID details to encode.
- * @param details.encryptionKey The DID encryption key.
- * @param details.serviceEndpoints The DID service endpoints.
+ * @param details.keyAgreement The DID encryption key.
+ * @param details.service The DID service endpoints.
  * @returns The Base58-encoded and CBOR-serialized off-chain DID optional details.
  */
 export function serializeAndEncodeAdditionalLightDidDetails({
-  encryptionKey,
-  serviceEndpoints,
-}: Pick<LightDidCreationDetails, 'encryptionKey' | 'serviceEndpoints'>):
-  | string
-  | null {
+  keyAgreement,
+  service,
+}: Pick<CreateDetailsInput, 'keyAgreement' | 'service'>): string | null {
   const objectToSerialize: SerializableStructure = {}
-  if (encryptionKey) {
-    objectToSerialize[ENCRYPTION_KEY_MAP_KEY] = encryptionKey
+  if (keyAgreement) {
+    const key = keyAgreement[0]
+    objectToSerialize[KEY_AGREEMENT_MAP_KEY] = key
   }
-  if (serviceEndpoints && serviceEndpoints.length > 0) {
-    objectToSerialize[SERVICES_KEY_MAP_KEY] = serviceEndpoints
+  if (service && service.length > 0) {
+    objectToSerialize[SERVICES_MAP_KEY] = service.map(({ id, ...rest }) => ({
+      id: stripFragment(id),
+      ...rest,
+    }))
   }
 
   if (Object.keys(objectToSerialize).length === 0) {
     return null
   }
 
+  const serializationFlag = 0x0
   const serialized = cborEncode(objectToSerialize)
-  // Add a flag to recognize the serialization algorithm. (Currently only custom object + cbor)
```

Comment:
I would keep the comment about `0x0` specifying our custom object + cbor, since I think it is not documented anywhere else 😅

Reply:
I am a big fan of the code that explains itself. Could a better name of this constant make it self-explanatory? I reused `serializationFlag` because this is what it is called in the `decode...` function. How about `versionOfEncoding` or `encodingVersion` or `serializationAlgId`?

Reply:
yes, naming would also help very much

Reply:
Done in b603395c